scripts/quests/bastok/Hyper_Active.lua

```diff
@@ -0,0 +1,159 @@
+-- Cerment Door: !gotoid 17531158
+-----------------------------------
+require('scripts/globals/common')
+require('scripts/globals/items')
+require('scripts/globals/keyitems')
+require('scripts/globals/npc_util')
+require('scripts/globals/quests')
+require('scripts/globals/titles')
+require('scripts/globals/zone')
+require('scripts/globals/interaction/quest')
+-----------------------------------
+local lowerDelkID = require('scripts/zones/Lower_Delkfutts_Tower/IDs')
+local lowerJeunoID = require('scripts/zones/Lower_Jeuno/IDs')
+-----------------------------------
+local quest = Quest:new(xi.quest.log_id.BASTOK, xi.quest.id.bastok.HYPER_ACTIVE)
+
+quest.reward =
+{
+    fame = 50,
+    fameArea = xi.quest.fame_area.BASTOK,
+    gil = 3000,
+}
+
+local spawnedMobs =
+{
+    lowerDelkID.mob.ORNA,
+    lowerDelkID.mob.FORMORIAN_SPEAR_1,
+    lowerDelkID.mob.FORMORIAN_SPEAR_2,
+
+}
+local examineLamp = function(player, npc)
+    local lampID = npc:getID()
+    local correctLampID = quest:getVar(player, "Lamp")
+
+    if lampID == correctLampID then
+        if quest:getVar(player, 'Prog') == 3 then
+            quest:setVar(player, 'Prog', 4)
+            quest:setVar(player, 'Lamp', 0)
+            npcUtil.giveKeyItem(player, xi.ki.HYPER_ALTIMETER)
+            return quest:messageSpecial(lowerJeunoID.text.HYPER_ALTIMETER, 0, xi.ki.HYPER_ALTIMETER, xi.ki.MOLYBDENUM_BOX)
+        else
+        end
+    end
+end
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+            player:hasCompletedQuest(xi.quest.log_id.BASTOK, xi.quest.id.bastok.TEAK_ME_TO_THE_STARS) and
+            not quest:getMustZone(player)
+        end,
+
+        [xi.zone.METALWORKS] =
+        {
+            ['Raibaht'] = quest:progressEvent(866, 0, xi.ki.HYPER_ALTIMETER, xi.ki.MOLYBDENUM_BOX),
+
+            onEventFinish =
+            {
+                [866] = function(player, csid, option, npc)
+                    if option == 1 then
+                        quest:begin(player)
+                        quest:setVar(player, 'Prog', 1)
+                        npcUtil.giveKeyItem(player, xi.ki.MOLYBDENUM_BOX)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.METALWORKS] =
+        {
+            ['Raibaht'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 4 then
+                        return quest:progressEvent(867)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [867] = function(player, csid, option, npc)
+                    quest:complete(player)
+                    player:delKeyItem(xi.ki.MOLYBDENUM_BOX)
+                    player:delKeyItem(xi.ki.HYPER_ALTIMETER)
```

Comment:
Always safe to use the following as much as possible:
```suggestion
                    if quest:complete(player) then
                        player:delKeyItem(xi.ki.MOLYBDENUM_BOX)
                        player:delKeyItem(xi.ki.HYPER_ALTIMETER)
                    end
```

Reply:
I keep forgetting about the if complete - did remember in the one I'm finishing up now though!

Reply:
Either way I need to either completely redo this one or adjust/convert community service as I needed to change the lua script/db name to streetlamp